tests/system/test_system.py

```diff
@@ -1234,6 +1234,27 @@ def test_compose_replace_existing_blob(self):
         composed = original.download_as_string()
         self.assertEqual(composed, BEFORE + TO_APPEND)
 
+    def test_compose_with_generation_match(self):
```

Comment:
If I'm not mistaken, the behavior of this test would be identical if the generation match and metageneration match flags were not set. Can we additionally test for the case of a mismatch and verify that the flags are doing something (causing the request to error out, presumably)?